source/agora/common/BlockStorage.d

```diff
@@ -32,6 +32,73 @@ import std.mmfile;
 import std.path;
 import std.stdio;
 
+/*******************************************************************************
+
+    Define the Storage for Blocks
+
+*******************************************************************************/
+
+public interface IBlockStorage
+{
+    @safe nothrow :
+
+    /***************************************************************************
+
+        Returns:
+            the highest block
+
+    ***************************************************************************/
+
+    public const(Block)* getLastBlock ();
+
+
+    /***************************************************************************
+
+        Save block to the file.
```

Comment:
This could just say `Save block to the storage`, since it doesn't require it to be a file.